src/main/java/bio/terra/service/resourcemanagement/google/GoogleProjectService.java

```diff
@@ -164,6 +166,9 @@ private GoogleProjectResource newProject(
         Map<String, List<String>> roleIdentityMapping)
         throws InterruptedException {
 
+        // Ensure that the project Id is valid
+        validateProjectId(requestedProjectId);
```

Comment:
Not sure what the current conventions are, but my preference is for `validateXXX` methods to return true/false or some other validation summary and `ensureXXX` for methods that throw on validation errors. The idea of ensure is that the code won't progress if the validation fails. 

And if it was called `ensureValidProjectId` then you might not need a comment either.

Reply:
Ah fantastic point.  I totally agree.  I didn't love not using a boolean for a validate but I liked the idea of using Preconditions to be able to get a little more context.  I'll update

Reply:
(To be clear, I'll update to be `ensureValidProjectId`